dragonball-common/src/main/kotlin/me/vendoor/dragonball/common/database/MongoExtension.kt

```diff
@@ -0,0 +1,55 @@
+package me.vendoor.dragonball.common.database
+
+import com.mongodb.ConnectionString
+import com.mongodb.MongoClientSettings
+import com.mongodb.client.MongoClient
+import com.mongodb.client.MongoClients
+import com.mongodb.client.MongoCollection
+import com.mongodb.client.MongoDatabase
+import org.bson.Document
+import org.bson.codecs.configuration.CodecRegistries.fromProviders
+import org.bson.codecs.configuration.CodecRegistries.fromRegistries
+import org.bson.codecs.configuration.CodecRegistry
+import org.bson.codecs.pojo.PojoCodecProvider
+
+fun openClient(connectionString: String): MongoClient {
+    val pojoCodecRegistry: CodecRegistry = fromRegistries(MongoClientSettings.getDefaultCodecRegistry(),
+            fromProviders(PojoCodecProvider.builder().automatic(true).build()))
+
+    val settings = MongoClientSettings.builder()
+            .applyConnectionString(ConnectionString(connectionString))
+            .codecRegistry(pojoCodecRegistry)
+            .build()
+
+    return MongoClients.create(settings)
+}
+
+fun MongoClient.hasDatabase(name: String) =
+        this.listDatabaseNames()
+                .into(ArrayList())
+                .contains(name)
+
+fun MongoClient.getDatabaseIfExists(name: String) =
+        if (this.hasDatabase(name)) {
+            this.getDatabase(name)
+        } else {
+            null
```

Comment:
Wouldn't some exception be better instead of null?

Reply:
Discussed offline.